### PR TITLE
Changed to snapshot version; added maven repository for deployment

### PR DESCRIPTION
**Pull request by AHeise**
Added Maven Repo support (https://oss.sonatype.org/content/repositories/snapshots/eu/stratosphere/)
Snapshots are not pushed to Maven Central!

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -374,6 +386,28 @@
 					</classpathContainers>
 				</configuration>
 			</plugin>
+
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-enforcer-plugin</artifactId>
+				<version>1.3.1</version>
```
Comment (AHeise) [inline pom.xml:393]: Override oss-parent pom with the newest version of the enforcer plugin - the default version was buggy in my setup and I could not package/install without the artifacts being in the Maven Repo.

File: pom.xml [hunk 2]
```diff
@@ -2,11 +2,17 @@
 <project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
 	xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
 
+	<parent>
+		<groupId>org.sonatype.oss</groupId>
```
Comment (AHeise) [inline pom.xml:6]: From the official guide https://docs.sonatype.org/display/Repository/Sonatype+OSS+Maven+Repository+Usage+Guide

File: stratosphere-dist/pom.xml [hunk 3]
```diff
@@ -112,6 +112,14 @@
 					</descriptors>
 				</configuration>
 			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
```
Comment (AHeise) [inline stratosphere-dist/pom.xml:116]: Omit the deployment of the (large) dist files. Remove if wanted.

**Comment by rmetzger** [comment]
Hi @AHeise,
I'm currently merging the change into the main code line. But travis was unable to upload to sonatype:

```
[INFO] --- maven-deploy-plugin:2.7:deploy (default-deploy) @ ozone ---
Downloading: https://oss.sonatype.org/content/repositories/snapshots/eu/stratosphere/ozone/0.4-ozone-SNAPSHOT/maven-metadata.xml
Uploading: https://oss.sonatype.org/content/repositories/snapshots/eu/stratosphere/ozone/0.4-ozone-SNAPSHOT/ozone-0.4-ozone-20131001.081749-1.pom
Oct 01, 2013 8:17:49 AM org.apache.commons.httpclient.auth.AuthChallengeProcessor selectAuthScheme
INFO: basic authentication scheme selected
Oct 01, 2013 8:17:50 AM org.apache.commons.httpclient.HttpMethodDirector processWWWAuthChallenge
INFO: Failure authenticating with BASIC 'Sonatype Nexus Repository Manager'@oss.sonatype.org:443
```

https://travis-ci.org/rmetzger/ozone/jobs/11997775

Do you have any ideas?

This is the .travis.yml file I was using https://github.com/rmetzger/ozone/blob/AHeise-master/.travis.yml

**Comment by AHeise** [comment]
Hi Robert,

sounds like an authentication issue. My last ozone built succeeded with the upload but there are some factors that could change that.

The script relies on the travis encryption and it’s very likely that the private/public keys do not match. http://about.travis-ci.org/docs/user/encryption-keys/

There are several solutions.

a)      The best solution would be if I add some dima colleague to the sonatypes repo, so that they can setup their own password with travis encryption with the official dimalabs key. For this option, please follow step 2 in the guide https://docs.sonatype.org/display/Repository/Sonatype+OSS+Maven+Repository+Usage+Guide#SonatypeOSSMavenRepositoryUsageGuide-2.Signup

b)      You could also send me your public key and I try to generate the correct encrypted version of my username/password.

c)       You could tell me when the version is ready and I pull it into my repo, where everything is properly configured.

Please note, that until you change the version to a non-SNAPSHOT version, the artifacts are not uploaded to maven central. They would only be available at https://oss.sonatype.org/content/repositories/snapshots/eu/stratosphere/

Best,

Arvid

**Comment by rmetzger** [comment]
Merged manually in https://github.com/dimalabs/ozone/commit/bbfbc4bac3be5349f6ac8a9e33ee691d85226c91
Thanks to @AHeise  for the help with Travis and Sonatype (maven)

**Pull request merged**